### PR TITLE
Make repositories work with data in, entities out

### PR DESCRIPTION
**Pull request by AlfonsoUceda**
This PR implements issue #291 

### TODO
- [ ] Update doc

### CODE REVIEW

File: lib/hanami/entity.rb [hunk 1]
```diff
@@ -219,9 +219,11 @@ def ==(other)
     #
     #   user = User.new(id: 23, name: 'Luca')
     #   user.to_h # => { :id => 23, :name => "Luca" }
+    #   user.to_hash # => { :id => 23, :name => "Luca" }
     def to_h
       Hash[attribute_names.map { |a| [a, read_attribute(a)] }]
```
Comment (joneslee85) [inline lib/hanami/entity.rb]: Should we use `Hanami::Utils::Hash`?

File: test/integration/repository_test.rb [hunk 2]
```diff
@@ -107,6 +107,18 @@
...
           end
+
+          it 'works with entity' do
+            article = ArticleRepository.new.persist(Article.new(title: 'Art 1'))
+            article.must_be_kind_of Article
+            article.id.wont_be_nil
+          end
+
+          it 'works with hash' do
+            article = ArticleRepository.new.persist({ title: 'Art 1' })
```
Comment (joneslee85) [inline test/integration/repository_test.rb]: worth to have a test for string key hash

File: lib/hanami/entity.rb (hunk 1 above)
Comment (jodosha) [inline lib/hanami/entity.rb]: @joneslee85 This won't make `#to_h` compatible with the rest of Ruby ecosystem, which doesn't know about our `Utils::Hash`.

File: test/integration/repository_test.rb (hunk 2 above)
Comment (jodosha) [inline test/integration/repository_test.rb]: Yes both :+1:

**Comment by AlfonsoUceda** [comment]
I had problems with rebase, I'll create another PR when this is done.

File: lib/hanami/entity.rb [hunk 3]
```diff
@@ -219,9 +219,11 @@ def ==(other)
...
     end
+    alias :to_hash :to_h
```
Comment (AlfonsoUceda) [inline lib/hanami/entity.rb]: @jodosha I think we can remove this changes now, what do you think?

Comment (jodosha) [inline lib/hanami/entity.rb]: @AlfonsoUceda Yes. :+1:

**Comment by jodosha** [comment]
@AlfonsoUceda Closing this in favor of a new implementation I'm working on. Thanks!

**Pull request closed without merging**